spec/integration/util/windows/user_spec.rb

```diff
@@ -155,6 +155,21 @@ def expect_logon_failure_error(&block)
       end
     end
 
+    describe 'logon_user_by_logon_type' do
+      let(:password){'test'}
+      let(:user){'test'}
+      let(:fLOGON32_LOGON_INTERACTIVE){2}
+      let(:fLOGON32_LOGON_NETWORK){3}
+      let(:fLOGON32_PROVIDER_DEFAULT){0}
+      let(:token){'test_token'}
+      it 'should raise error given that logon returns false' do
+        allow(Puppet::Util::Windows::User).to receive(:LogonUserW).and_return(false)
+        expect(Puppet::Util::Windows::User.logon_user_by_logon_type(user, password, fLOGON32_LOGON_INTERACTIVE, fLOGON32_PROVIDER_DEFAULT, token))
+            .to raise_error(Puppet::Util::Windows::Error.new(_("Failed to logon user %{name}") % { name: name.inspect }))
```

Comment:
I think this is failing because it's much too literally taken from line 88 of user.rb. The parameter to `raise_error` should be the result of all of the function calls on 88, not the function calls themselves, so it should should probably be something like `raise_error("Failed to logon user test")`